### PR TITLE
Sign Up configuration and Fixed errors in Post Model

### PR DESCRIPTION
**Pull request by BlazeDex**
# Description

Implemented ActionMailer with SendGrid to send the e-mails confirmations.
Added confirmable validation on User model.
Added an input to save user's name.
Added an input to save user's lastname.
Fixed bug, the current user cannot see posts from other users.

## Screenshots
Captura desde 2024-03-14 19-21-59

Captura desde 2024-03-14 19-22-23

Captura desde 2024-03-14 19-22-54

### CODE REVIEW

**Review by MauricioNC — APPROVED** [review]

File: Gemfile [hunk 1]
```diff
@@ -80,6 +80,8 @@ gem 'image_processing', '>= 1.2'
 
 gem 'dotenv', groups: %i[development test]
 
+gem 'jquery-rails'
+gem 'jquery-ui-rails'
```
Comment (juperameza) [inline Gemfile]: Why are you using jQuery?

**Review by juperameza — CHANGES_REQUESTED** [review]

File: Gemfile (hunk 1 above)
Comment (BlazeDex) [reply Gemfile]: Why not? 🤓

Nah, I forgot delete that gems. I was using different methods to send the emails before use SendGrid (like MailCatcher and others). Thx for the review.

File: config/environment.rb [hunk 2]
```diff
@@ -1,5 +1,15 @@
 # Load the Rails application.
-require_relative "application"
+require_relative 'application'
 
 # Initialize the Rails application.
 Rails.application.initialize!
+
+ActionMailer::Base.smtp_settings = {
+  user_name: 'apikey', # This is the string literal 'apikey', NOT the ID of your API key
+  password: ENV['SENDGRID_API_KEY'], # This is the secret sendgrid API key which was issued during API key creation
```
Comment (JuanequeX) [inline config/environment.rb:9]: Sometimes these types of `env vars` integrations are not placed in the `.env` file. Instead, consider configuring your Rails application environments using `Rails.application.credentials` via `RAILS_MASTER_KEY` for access. In fresh Rails apps, this configuration is essential.

Comment (JuanequeX) [reply config/environment.rb:9]: FYI, this is only info, not a request for changes. If you want to learn more about it. look at these:
- [resource](https://github.com/rails/rails/pull/33521)
- [resource](https://discuss.rubyonrails.org/t/rails-master-key-and-per-environment-init/82615/2?u=juanequex)
- [resource](https://medium.com/@adi8090808766/credentials-or-environment-variables-in-ruby-on-rails-371831f2502)
- [resource](https://dev.to/raaynaldo/the-power-of-rails-master-key-36fh)

**Review by JuanequeX — APPROVED** [review]

File: config/environment.rb (hunk 2 above)
Comment (BlazeDex) [reply config/environment.rb:9]: Thx!

**Review by juperameza — APPROVED** [review]

**Pull request merged**